Interop/Indexer/prebuilt/nativeInteropStubs/cpp/signalChaining.cpp

```diff
@@ -0,0 +1,58 @@
+#if defined(__linux__) || defined(__APPLE__)
+#include <dlfcn.h>
+#include <stdio.h>
+#include <signal.h>
+#if defined(__linux__)
+#include <link.h>
+#endif
+
+typedef void (*toggleCrashRecovery_t)(int);
+typedef int (*sigaction_t)(int, const struct sigaction *, struct sigaction *);
+
+static sigaction_t realSigaction = 0;
+
+static int mySigaction(int sig, const struct sigaction *act, struct sigaction * oact) {
+  if (sig == SIGILL || sig == SIGFPE|| sig ==  SIGSEGV || sig == SIGBUS || sig == SIGUSR1|| sig == SIGUSR2) return 0;
+  return realSigaction(sig, act, oact);
+}
+
+__attribute__((constructor))
+static void initSignalChaining() {
+  toggleCrashRecovery_t clang_sym = 0;
+  void** base = 0;
+  Dl_info info;
+
+  clang_sym = (toggleCrashRecovery_t)dlsym(RTLD_DEFAULT, "clang_toggleCrashRecovery");
```

Comment:
IIRC, this code is linked with `libclang` and can use its headers, so `clang_toggleCrashRecovery` can be accessed directly.

Reply:
Yeah,, but it doesn't change much.

Reply:
It can make this code simpler.

Reply:
Well, this way we guarantee that symbol found by the default lookup is in dynamic library we want it to be. Also I kind of like this code being pretty universal wrt symbols it patches.